pyfstat/helper_functions.py

```diff
@@ -880,3 +882,19 @@ def predict_fstat(
     twoF_sigma = float(d["twoF_sigma"])
     os.remove(tempory_filename)
     return twoF_expected, twoF_sigma
+
+
+def parse_list_of_numbers(val):
+    """Convert a number, list of numbers or comma-separated str into a list of numbers.
+
+    This is useful e.g. for sqrtSX inputs where the user can be expected
+    to try either type of input.
+    """
+    try:
+        out = list(map(float, val.split(",")))
+    except Exception:
+        try:
+            out = list(map(float, np.atleast_1d(val)))
+        except Exception:
+            raise ValueError(f"Could not parse '{val}' as a number or list of numbers.")
+    return out
```

Comment:
I feel these `try`s should be more explicit, but I don't know how...
Maybe 
```suggestion
    try:
        out = list(map(float, val.split(",") 
        if hassattr(val, "split")
        else np.atleast_1d(val)) 
    except Exception as e:
        raise ValueError(f"Could not parse '{val}' as a number or list of numbers: {e.message}")
```

Reply:
That doesn't look black-legal, but I like the if-else simplification. Will try it locally.

Reply:
I'd bet it's not (I'm actually quite surprised it came out with the "right" identation :upside_down_face: )

Reply:
Resolved, but just returning `e` not `e.message`.